### PR TITLE
Adds new mechanics to AI/stationbound control of airlocks - allows AIs to bolt airlocks again, sort of

### PR DESCRIPTION
**Pull request by Amunak**
This adds new mechanics around AIs being able to bolt doors (as well as other related minor improvements). [[Feedback thread](https://forums.aurorastation.org/topic/15046-feedback-new-mechanics-around-ais-and-stationbounds-bolting-airlock-doors/)]

### Goals
I wanted to return the ability for AIs to bolt doors again, as it's a logic, fun mechanic, but there are good reasons ([original forum thread](https://forums.aurorastation.org/topic/14362-stationbound-nerfs/), [recent forum thread](https://forums.aurorastation.org/topic/14465-limit-ais-ability-to-be-extension-of-security/)) for why it was removed. There are also good reasons ([forum thread](https://forums.aurorastation.org/topic/14904-ais-round-start-remote-mech-removal-feedback/)) to give AIs more things to do.

The goal was to return bolting mechanics to AIs and stationbounds but keeping this in mind:
* The implementation should be logical and consistent. Stuff that *makes sense*.
* It should be easily configurable so that we can tweak the balance easily.
* It should be well integrated into current AI mechanics (and how people are used to bolting working). Nothing outrageously new and different.
* It should add options both to AI players *and* everyone else, make it more fun.
* Obviously it should not be as powerful as "standard bolting" was, otherwise we'd be just reintroducing the issue.

### Implementation
I (hopefully) achieved the above goals with these new and changed features:
* Every airlock now has a toggle telling whether AIs can bolt it.
* There is now an announcement and a delay of 8 seconds for dropping and 4 seconds for raising the bolts by AIs (though traitor AIs are not affected by either - unless they explicitly choose to). That is still more than enough time to even bolt open doors, as it takes 15 seconds for doors to close. Some door types override this (high security have 10/5 and vault has 12/8 - the reasoning being that they are more "heavy duty" and thus take longer). After doing either the bolt/unbolt request cannot be re-sent by any AI/borg until complete.
* Players can "pulse" the "AI control wire" (when hacking the door) to toggle this.
  * In the hacking UI, the AI control light has several states instead of just on or off. It can be 'off' (cut), 'red' (emagged), 'orange' (AI can control doors but not bolt them) or 'green' (AI can bolt or unbolt the doors).
  * This makes pulsing the wire actually useful for something other than identifying the AI control wire.
* This "AI can bolt" option on airlocks has several configurations (that can be overriden on a subtype or when mapping):
  * By default it is set to auto-configuration. Autoconf sets the "AI can bolt" to `false` whent the airlock starts unlocked and to `true` when it starts locked. Pulsing the AI control wire can change this.
  * External airlocks and hatches (those are on tcommsat for example) are boltable by default, but can be toggled by pulsing the AI control wire.
  * High security airlocks are set to `allow`, which means the AIs can bolt these *and* it cannot be overridden by pulsing the AI control wire. Of course you can just cut the wire to disable all AI control for the airlock.
  * Centcom airlocks are set to `never`, which does not allow even traitor AIs to control them.
* As a side effect of these changes AIs can no longer unbolt any random bolted doors as it was implemented in #8889. However due to the above defaults they can still unbolt any doors bolted at round start, so this actually adds a form of defense against AIs - you can bolt the doors (using hacking, probably) and they won't be able to unbolt them unless you also allow them to do so by pulsing the AI control wire.

Overall I believe the defaults make sense, it still allows AIs to somewhat protect secure areas, but they can't just lock traitors in maintenance or just anywhere. It also adds some challenge to getting into the more secure areas, while giving bigger incentive for traitors to subvert AIs.

For reference this is a screenshot of engineering showing which doors would now be boltable by AIs:
Engineering with boltable doors

Airlocks, engine room, engine access hatch and secure storage. Somewhere like on the sublevel near AI Core and Bunker almost every single door would be boltable (except for elevator doors and the south maintenance door).

### Related changes, tweaks and bugfixes
* Hacking UI now shows pretty colors and has tweaked messages.
* AI players have a new "AI Help" verb. Most messages that AI players recieve when they join have been moved there, along with new OOC notes that instruct them to keep in mind their power and not to ruin other peoples' fun. 
* The bolt announcement message is now audible instead of visual. It does not show up (and the bolt/unbolt request doesn't go through) when the door is unpowered.
* Added some fancy messages for when the doors are unpowered and the AI cannot open/close/(un)bolt them.
* AIs, when using 'Radio Settings', now see their radio channel keys. They are also displayed in 'AI-Help'.
* Moved code that gathers radio channels for AI to a separate proc.
* Fixed admin check in airlock VueUI. Also allows admins to actually toggle stuff like bolts, safeties and electrification. They can do so "in secret" and mimic an AI, dropping/raising the bolts with the announcement. They do skip some checks (like cut bolt wire) when they force toggle it though.
* Door UI now has the area name and name of the door.
* Malfunctioning / traitor AIs and borgs can pick from either instant (un)bolting (either by ctrl+clicking or by using a new "bolts override" option in the UI), or they can "fake" a non-traitor AI by using the dialogue and the regular raise/drop buttons.
* Refactored some of the airlock code, especially stuff regarding displaying messages and toggling bolts. Some of it was necessary for these changes, other parts I think I just made more readable.
* Locked AI core on Runtime, for easier testing and also because there otherwise wouldn't be anything locked on the way to the AI there. Realism!

### Points of contention / tweaking [[Feedback thread](https://forums.aurorastation.org/topic/15046-feedback-new-mechanics-around-ais-and-stationbounds-bolting-airlock-doors/)]
* If this is deemed as too much of a buff and return to previous state, we can always tweak the type of doors AIs can bolt by default. I'm not so sure about the high security doors - what do you think?
* Waiting 8 seconds for bolts to drop might be too much. But it definitely allows reaction from people; if the AI was trying to lock you in you could just open the doors and they'd bolt open.
* The "AI Help" text, especially the OOC part might be too wordy. I'm open to suggestions.

#### Airlock admin UI (also same for malf/traitors):
Airlock Admin/Traitor/Malf UI

#### Limit on spamming (un)bolting:
dreamseeker_2020-10-20_21-18-05

#### From outsiders' point of view (this takes place in over 24 seconds):
dreamseeker_2020-10-20_21-18-13

#### AI join messages and AI Help pop-up; Also showing `Radio-Settings` message.
dreamseeker_2020-10-21_12-17-31

### CODE REVIEW

**Comment by Chada-1** [comment]
I'd be open to the AI being able to bolt only certain high secure areas like the vault/AI Core/Command

**Comment by Amunak** [comment]
> I'd be open to the AI being able to bolt only certain high secure areas like the vault/AI Core/Command
> Edit: O I C, so that's the plan anyhoo

Yep; that's how it would work out. I clarified it and added an example screenshot with enginnering showing what exactly would be now boltable.

**Comment by Amunak** [comment]
!review

**Review by Chada-1 — APPROVED** [review]

**Review by Arrow768 — CHANGES_REQUESTED** [review]
The current implementation allows players to meta game malf/traitor APIs (and borgs?) by ordering them to bolt a airlock.
If this action is instant when it should not be, it is a certain that the AI is hostile.

I would suggest to add a button to the UI that instantly bolts/unbolted a airlock to avoid this situation.

**Comment by Amunak** [comment]
Updates (also reflected in original description)! Please re-review the topic conditionals, they changed quite a bit.

- Added "designation" to the Airlock UI that shows the area and name of the airlock.
- Malf/Traitor AIs and admins can choose to use the "slow", announced bolting and unbolting in order not to appear suspicious (and hopefully prevent people from OOC guessing that it's a malf AI). When doing this, both admins and traitors are still limited by the checks like AI control wire so that they cannot accidentally reveal themselves there. Ctrl+clicking always uses the "override" method because speed is usually wanted there - for stealth they can do that one extra click.
- refactor: Moved Initialize and Destroy to beginning of the file because it's hard to find otherwise, and merged with the second Initialize I added by mistake.
- refactor: Changed some of the VueUI variables to make them readable. It was a mess. It's still a mess but at least you can tell what is what.

Screenshot of the updated UI:
Airlock Admin/Traitor/Malf UI

**Comment by Amunak** [comment]
Just for reference I confirmed that:
- Borgs work as expected - behave same as AIs, including traitor borgs.
- Carded AIs also work as expected, when wireless is enabled they have the door UI like usual with all the new upgrades.
- pAIs are unaffected, they use something completely different.

**Comment by Amunak** [comment]
* Reworked the VueUI _once more_. The Raise/Drop Now button is now just only one button next to the regular bolt controls. Looks nicer, is probably less confusing, and I'd have to change the template regardless: bolts are no longer a part of the JS define there. It greatly simplified that whole UI part though.
* Found a few bugs that have been squashed. In particular it's no longer possible to "queue" a bolt/unbolt action while the doors are unpowered. The silicons now get a nice message about this.
* I separated the bolt speed and unbolt speed, made the unbolting generally twice as fast as bolting. Most doors stay at 8sec bolting and 4sec unbolting, but high security doors are up to 10/5 and vault is 12/8. Let's say that these doors are heavier, with sturdier bolts, and thus take longer to operate.
* Added some fancy messages for when the doors are unpowered and the AI cannot open/close/(un)bolt them.

Even newer airlock admin UI

**Review by JohnWildkins — APPROVED** [review]
